src/storage/MonitoringStore.ts

```diff
@@ -29,31 +31,44 @@ export class MonitoringStore<T extends ResourceStore = ResourceStore>
   }
 
   public async addResource(container: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier> {
-    const identifier = await this.source.addResource(container, representation, conditions);
-    this.emitChanged([ container, identifier ]);
-    return identifier;
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
+    return this.emitChanged(await this.source.addResource(container, representation, conditions));
   }
 
   public async deleteResource(identifier: ResourceIdentifier,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
     return this.emitChanged(await this.source.deleteResource(identifier, conditions));
   }
 
   public async setRepresentation(identifier: ResourceIdentifier, representation: Representation,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
     return this.emitChanged(await this.source.setRepresentation(identifier, representation, conditions));
   }
 
   public async modifyResource(identifier: ResourceIdentifier, patch: Patch,
-    conditions?: Conditions): Promise<ResourceIdentifier[]> {
+    conditions?: Conditions): Promise<Record<string, RepresentationMetadata>> {
     return this.emitChanged(await this.source.modifyResource(identifier, patch, conditions));
   }
 
-  private emitChanged(identifiers: ResourceIdentifier[]): typeof identifiers {
-    for (const identifier of identifiers) {
-      this.emit('changed', identifier);
+  private emitChanged(changes: Record<string, RepresentationMetadata>): Record<string, RepresentationMetadata> {
+    for (const [ key, value ] of Object.entries(changes)) {
+      const activity = value.get(SOLID_AS.terms.Activity)?.value;
```

Comment:
Just realized this is going to error if there are multiple values for Activity. It can also potentially send invalid values in case the source store set a wrong value there. There should probably be checks for these cases to make sure this store only emits correct values.